charts/opentelemetry-collector/templates/_config.tpl

```diff
@@ -855,9 +855,16 @@ processors:
     profile_statements:
     # prioritized by
     # https://opentelemetry.io/docs/specs/semconv/non-normative/k8s-attributes/#how-servicename-should-be-calculated
-      - set(resource.attributes["service.name"], resource.attributes["service.name"])
-        where resource.attributes["service.name"] != nil
+     {{- range $index, $serviceAnnotation := .Values.presets.profilesCollection.serviceAnnotations }}
+      - set(resource.attributes["service.name"], resource.attributes[{{ $serviceAnnotation.tag_name | quote }}])
+        where resource.attributes["service.name"] == nil and resource.attributes[{{ $serviceAnnotation.tag_name | quote }}] != nil
 
+    {{- end }}
+    {{- range $index, $serviceLabel := .Values.presets.profilesCollection.serviceLabels }}
+      - set(resource.attributes["service.name"], resource.attributes[{{ $serviceLabel.tag_name  | quote }}])
+        where resource.attributes["service.name"] == nil and resource.attributes[{{ $serviceLabel.tag_name | quote }}] != nil
+
+    {{- end }}
```

Comment:
I'm not sure I get what's going on here. It doesn't seem to make a lot of sense to me. Potentially related to the example that didn't rendered (see my other comment).

From what I understand, you loop over all service annotations and labels and set the `service.name` attribute to their value? Won't this overwrite the `service.name` attribute multiple times?

Reply:
Ahhh! Makes total sense. 👍